main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -199,10 +200,19 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
       val annVal = visitAnnotations(ann0, root)
       val tparams = getTypeParams(tparams0)
       val fparams = getFormalParams(fparams0, subst)
+      val returnType = getReturnType(sc.base, fparams.length)
```

Comment:
Instead of reconstructing it here, I was actually thinking of carrying it through from earlier phases. But I guess that could have consequences, so I am not sure.